### ISSUE TITLE
about multiple images

### ISSUE THREAD

**Issue by wyz1212** [jacobgil/pytorch-grad-cam#170]
Thank you for your excellent work!
I successfully read an image with the code you provided(resnet-50), you mentioned that the input_tensor can be a batch tensor with several images! But when I read multiple images (for example, 8 images) at one time, the code reports an error, and the environment configuration should be OK. What specific modifications should I make?

**Comment by jacobgil** [comment]
Hi,
I need more details to reproduce a bug or answer.
Are you passing target_category, or is it None?
What algorithm?
Can you post some code snippet to help understand or reproduce?

**Comment by wyz1212** [comment]
Because my task belongs to video classification, I need to add a time dimension to the model. I use the function image_concatenate1()to read in 8 images at the same time, and the input tensor dimension becomes [8, 3, 240, 320], instead of [1, 3, 240, 320], target_ Category = none, in order to match the input, grayscale_ cam = grayscale_ Cam [0, :] is rewritten as grayscale_ cam = grayscale_ Cam [:, :, :], so its dimension becomes (8, 240, 320) instead of (240, 320). The detailed code is as follows:
……
    target_layers = [model.layer4[-1]]

    input_tensor = image_concatenate()

    target_category = None

    cam_algorithm = methods[args.method]
    with cam_algorithm(model=model,
                       target_layers=target_layers,
                       use_cuda=args.use_cuda) as cam:

        cam.batch_size = 32

        grayscale_cam = cam(input_tensor=input_tensor,
                            target_category=target_category,
                            aug_smooth=args.aug_smooth,
                            eigen_smooth=args.eigen_smooth)

        rgb = image_concatenate2()
        grayscale_cam = grayscale_cam[:, :, :]  
        cam_image = show_cam_on_image(rgb, grayscale_cam, use_rgb=True)
……
def image_concatenate():
    rgb_img = cv2.imread(r'image_00001.jpg', 1)[:, :, ::-1]
    rgb_img = np.float32(rgb_img) / 255
    input_tensor = preprocess_image(rgb_img, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) 
    for dirname, _, filenames in os.walk(r'./1'):
        for filename in filenames:
            rgb_img = cv2.imread(os.path.join(dirname, filename), 1)[:, :, ::-1]
            rgb_img = np.float32(rgb_img) / 255
            tensor = preprocess_image(rgb_img, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
            input_tensor = torch.cat((input_tensor, tensor), dim=0) #torch.Size([8, 3, 240, 320])

    return input_tensor 

def image_concatenate2():
    x = cv2.imread(r'image_00001.jpg', 1)[:, :, ::-1]
    x = np.array(x)
    x = np.expand_dims(x, axis=0)  # (1, 240, 320, 3)
    for dirname, _, filenames in os.walk(r'./1'):
        for filename in filenames:
            im = cv2.imread(os.path.join(dirname, filename), 1)[:, :, ::-1]
            image_array = np.array(im)
            image_array = np.expand_dims(image_array, axis=0)
            x = np.concatenate((x, image_array), axis=0)
            x = np.float32(x) / 255  #(8, 240, 320, 3)

    return x

I tried some methods, but they didn't work. Please help me, Thank you very much! @jacobgil

**Comment by jacobgil** [comment]
Hi,
I think the issue is that show_cam_on_image expects to get a single image, and doesn't support a batch of images.
Try something like
```python
cam_images = [show_cam_on_image(img, grayscale, use_rgb=True) for img, grayscale in zip(rgb_img, grayscale_cam)]
```
